xontrib/brace_expansion.py

```diff
@@ -21,8 +21,12 @@
 
 @events.on_transform_command
 def expand_braces(cmd, **_):
+    if '[' in cmd or '$(' in cmd or '@(' in cmd:
+        return cmd
+
```

Comment:
Please don't sprawl the scope of the PR. I don't like this and won't accept it; whatever this is doing should be done with an ast, but more importantly, it should be another PR so we can discuss that separately.